gdal/frmts/rda/rdadataset.cpp

```diff
@@ -626,26 +653,47 @@ bool GDALRDADataset::ParseConnectionString( GDALOpenInfo* poOpenInfo )
     JsonObjectUniquePtr oObj(poObj);
 
     json_object* poGraphId =
-        CPL_json_object_object_get(poObj, "graph-id");
-    if( poGraphId == nullptr ||
-        json_object_get_type(poGraphId) != json_type_string )
+        json_object_object_get(poObj, "graph-id");
```

Comment:
please keep the use of CPL_json_object_object_get(). json_object_object_get() is deprecated by newer versions of libjson-c and will emit warnings

Reply:
done